// contentful-management/dist/es-modules/adapters/REST/endpoints/webhook.js
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
import copy from 'fast-copy';
import * as raw from './raw';
import { normalizeSelect } from './utils';
const getBaseUrl = params => `/spaces/${params.spaceId}/webhook_definitions`;
const getWebhookCallBaseUrl = params => `/spaces/${params.spaceId}/webhooks`;
const getWebhookUrl = params => `${getBaseUrl(params)}/${params.webhookDefinitionId}`;
const getWebhookCallUrl = params => `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/calls`;
const getWebhookCallDetailsUrl = params => `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/calls/${params.callId}`;
const getWebhookHealthUrl = params => `${getWebhookCallBaseUrl(params)}/${params.webhookDefinitionId}/health`;
const getWebhookSettingsUrl = params => `/spaces/${params.spaceId}/webhook_settings`;
const getWebhookSigningSecretUrl = params => `${getWebhookSettingsUrl(params)}/signing_secret`;
const getWebhookRetryPolicyUrl = params => `${getWebhookSettingsUrl(params)}/retry_policy`;
export const get = (http, params) => {
  return raw.get(http, getWebhookUrl(params));
};
export const getManyCallDetails = (http, params) => {
  return raw.get(http, getWebhookCallUrl(params), {
    params: normalizeSelect(params.query)
  });
};
export const getCallDetails = (http, params) => {
  return raw.get(http, getWebhookCallDetailsUrl(params));
};
export const getHealthStatus = (http, params) => {
  return raw.get(http, getWebhookHealthUrl(params));
};
export const getMany = (http, params) => {
  return raw.get(http, getBaseUrl(params), {
    params: normalizeSelect(params.query)
  });
};
export const getSigningSecret = (http, params) => {
  return raw.get(http, getWebhookSigningSecretUrl(params));
};
export const getRetryPolicy = (http, params) => {
  return raw.get(http, getWebhookRetryPolicyUrl(params));
};
export const create = (http, params, rawData, headers) => {
  const data = copy(rawData);
  return raw.post(http, getBaseUrl(params), data, {
    headers
  });
};
export const createWithId = (http, params, rawData, headers) => {
  const data = copy(rawData);
  return raw.put(http, getWebhookUrl(params), data, {
    headers
  });
};
export const update = /*#__PURE__*/function () {
  var _ref = _asyncToGenerator(function* (http, params, rawData, headers) {
    var _rawData$sys$version;
    const data = copy(rawData);
    delete data.sys;
    return raw.put(http, getWebhookUrl(params), data, {
      headers: _objectSpread({
        'X-Contentful-Version': (_rawData$sys$version = rawData.sys.version) !== null && _rawData$sys$version !== void 0 ? _rawData$sys$version : 0
      }, headers)
    });
  });
  return function update(_x, _x2, _x3, _x4) {
    return _ref.apply(this, arguments);
  };
}();
export const upsertSigningSecret = /*#__PURE__*/function () {
  var _ref2 = _asyncToGenerator(function* (http, params, rawData) {
    const data = copy(rawData);
    return raw.put(http, getWebhookSigningSecretUrl(params), data);
  });
  return function upsertSigningSecret(_x5, _x6, _x7) {
    return _ref2.apply(this, arguments);
  };
}();
export const upsertRetryPolicy = /*#__PURE__*/function () {
  var _ref3 = _asyncToGenerator(function* (http, params, rawData) {
    const data = copy(rawData);
    return raw.put(http, getWebhookRetryPolicyUrl(params), data);
  });
  return function upsertRetryPolicy(_x8, _x9, _x10) {
    return _ref3.apply(this, arguments);
  };
}();
export const del = (http, params) => {
  return raw.del(http, getWebhookUrl(params));
};
export const deleteSigningSecret = /*#__PURE__*/function () {
  var _ref4 = _asyncToGenerator(function* (http, params) {
    return raw.del(http, getWebhookSigningSecretUrl(params));
  });
  return function deleteSigningSecret(_x11, _x12) {
    return _ref4.apply(this, arguments);
  };
}();
export const deleteRetryPolicy = /*#__PURE__*/function () {
  var _ref5 = _asyncToGenerator(function* (http, params) {
    return raw.del(http, getWebhookRetryPolicyUrl(params));
  });
  return function deleteRetryPolicy(_x13, _x14) {
    return _ref5.apply(this, arguments);
  };
}();
